Virt-Handler frequently crashing with NPE

<!-- This form is for bug reports and feature requests ONLY!
Also make sure that you visit our User Guide at https://kubevirt.io/user-guide/
-->

**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
/kind bug
> /kind enhancement


**What happened**:

The virt-handler frequently crashes with a NPE for me:

```
/root/go/src/kubevirt.io/kubevirt/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88\n/gimme/.gimme/versions/go1.10.linux.amd64/src/runtime/asm_amd64.s:2361\n","pos":"runtime.go:66","timestamp":"2018-11-29T15:44:12.766976Z"}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x109fcb9]

goroutine 137 [running]:
kubevirt.io/kubevirt/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/root/go/src/kubevirt.io/kubevirt/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x107
panic(0x11f71e0, 0x1d32f50)
	/gimme/.gimme/versions/go1.10.linux.amd64/src/runtime/panic.go:505 +0x229
kubevirt.io/kubevirt/pkg/virt-handler.(*VirtualMachineController).hasGracePeriodExpired(0xc42038e000, 0xc420099800, 0x14c4020, 0xc4203e26b8)
	/root/go/src/kubevirt.io/kubevirt/pkg/virt-handler/vm.go:158 +0x59
kubevirt.io/kubevirt/pkg/virt-handler.(*VirtualMachineController).processVmShutdown(0xc42038e000, 0xc420ab18c0, 0xc420099800, 0x0, 0x3)
	/root/go/src/kubevirt.io/kubevirt/pkg/virt-handler/vm.go:975 +0xed
kubevirt.io/kubevirt/pkg/virt-handler.(*VirtualMachineController).defaultExecute(0xc42038e000, 0xc420c0c600, 0x12, 0xc420ab18c0, 0xc4202c5801, 0xc420099800, 0x1, 0xc4200cc000, 0x14a6a00)
	/root/go/src/kubevirt.io/kubevirt/pkg/virt-handler/vm.go:768 +0xcdb
kubevirt.io/kubevirt/pkg/virt-handler.(*VirtualMachineController).execute(0xc42038e000, 0xc420c0c600, 0x12, 0x118b160, 0xc420be79d0)
	/root/go/src/kubevirt.io/kubevirt/pkg/virt-handler/vm.go:872 +0x315
kubevirt.io/kubevirt/pkg/virt-handler.(*VirtualMachineController).Execute(0xc42038e000, 0xc420324c00)
	/root/go/src/kubevirt.io/kubevirt/pkg/virt-handler/vm.go:467 +0x10d
kubevirt.io/kubevirt/pkg/virt-handler.(*VirtualMachineController).runWorker(0xc42038e000)
	/root/go/src/kubevirt.io/kubevirt/pkg/virt-handler/vm.go:457 +0x2b
kubevirt.io/kubevirt/pkg/virt-handler.(*VirtualMachineController).(kubevirt.io/kubevirt/pkg/virt-handler.runWorker)-fm()
	/root/go/src/kubevirt.io/kubevirt/pkg/virt-handler/vm.go:449 +0x2a
kubevirt.io/kubevirt/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc420438030)
	/root/go/src/kubevirt.io/kubevirt/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x54
kubevirt.io/kubevirt/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc420438030, 0x3b9aca00, 0x0, 0x1, 0xc4204ee600)
	/root/go/src/kubevirt.io/kubevirt/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xbd
kubevirt.io/kubevirt/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc420438030, 0x3b9aca00, 0xc4204ee600)
	/root/go/src/kubevirt.io/kubevirt/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by kubevirt.io/kubevirt/pkg/virt-handler.(*VirtualMachineController).Run
	/root/go/src/kubevirt.io/kubevirt/pkg/virt-handler/vm.go:449 +0x558
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:


**Anything else we need to know?**:

**Environment**:
- KubeVirt version (use `virtctl version`):

```
$ virtctl version
Client Version: version.Info{GitVersion:"v0.10.0", GitCommit:"6ddf4f4e2db4924ac0725676db72bb78acc6cca2", GitTreeState:"clean", BuildDate:"2018-11-08T13:47:51Z", GoVersion:"go1.10", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{GitVersion:"v0.10.0", GitCommit:"6ddf4f4e2db4924ac0725676db72bb78acc6cca2", GitTreeState:"clean", BuildDate:"2018-11-08T13:47:51Z", GoVersion:"go1.10", Compiler:"gc", Platform:"linux/amd64"}
```

- Kubernetes version (use `kubectl version`):

```
Server Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.0", GitCommit:"0ed33881dc4355495f623c6f22e7dd0b7632b7c0", GitTreeState:"clean", BuildDate:"2018-09-27T16:55:41Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"linux/amd64"}
```

- VM or VMI specifications:

```
apiVersion: kubevirt.io/v1alpha2
kind: VirtualMachineInstance
metadata:
  annotations:
    virtualmachinepreset.kubevirt.io/medium: kubevirt.io/v1alpha2
  labels:
    kubevirt.io/size: medium
  name: testvm
  namespace: kube-system
spec:
  domain:
    cpu:
      cores: 2
    devices:
      disks:
      - disk:
          bus: virtio
        name: registrydisk
        volumeName: registryvolume
      - disk:
          bus: virtio
        name: cloudinitdisk
        volumeName: cloudinitvolume
    resources:
      requests:
        memory: 4096M

  - name: registryvolume
    registryDisk:
      image: docker.io/alvaroaleman/kubevirt-bionic:3
  - cloudInitNoCloud:
      secretRef:
        name: userdata
    name: cloudinitvolume
```
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

--- a/pkg/virt-launcher/virtwrap/api/converter.go
+++ b/pkg/virt-launcher/virtwrap/api/converter.go
@@ -546,10 +546,12 @@ func Convert_v1_VirtualMachine_To_api_Domain(vmi *v1.VirtualMachineInstance, dom
 	domain.Spec.Devices.Channels = append(domain.Spec.Devices.Channels, newChannel)
 
 	domain.Spec.Metadata.KubeVirt.UID = vmi.UID
+	gracePeriodSeconds := v1.DefaultGracePeriodSeconds
 	if vmi.Spec.TerminationGracePeriodSeconds != nil {
-		domain.Spec.Metadata.KubeVirt.GracePeriod = &GracePeriodMetadata{
-			DeletionGracePeriodSeconds: *vmi.Spec.TerminationGracePeriodSeconds,
-		}
+		gracePeriodSeconds = *vmi.Spec.TerminationGracePeriodSeconds
+	}
+	domain.Spec.Metadata.KubeVirt.GracePeriod = &GracePeriodMetadata{
+		DeletionGracePeriodSeconds: gracePeriodSeconds,
 	}
 
 	domain.Spec.SysInfo = &SysInfo{}

--- a/tests/vmi_lifecycle_test.go
+++ b/tests/vmi_lifecycle_test.go
@@ -1001,11 +1001,15 @@ var _ = Describe("VMIlifecycle", func() {
 			})
 		})
 		Context("with ACPI and some grace period seconds", func() {
-			It("should result in vmi status succeeded", func() {
-
+			table.DescribeTable("should result in vmi status succeeded", func(gracePeriod int64) {
 				vmi = newCirrosVMI()
-				gracePeriod := int64(10)
-				vmi.Spec.TerminationGracePeriodSeconds = &gracePeriod
+
+				if gracePeriod >= 0 {
+					vmi.Spec.TerminationGracePeriodSeconds = &gracePeriod
+				} else {
+					gracePeriod = v1.DefaultGracePeriodSeconds
+					vmi.Spec.TerminationGracePeriodSeconds = nil
+				}
 
 				By("Creating the VirtualMachineInstance")
 				obj, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
@@ -1023,7 +1027,10 @@ var _ = Describe("VMIlifecycle", func() {
 					Expect(err).ToNot(HaveOccurred(), "Should get VMI")
 					return currVMI.Status.Phase
 				}, gracePeriod+5, 0.5).Should(Equal(v1.Succeeded), "VMI should be succeeded")
-			})
+			},
+				table.Entry("with set grace period seconds", int64(10)),
+				table.Entry("with default grace period seconds", int64(-1)),
+			)
 		})
 		Context("with grace period greater than 0", func() {
 			It("should run graceful shutdown", func() {

